tests/templates/spark-pi.yaml

```diff
@@ -0,0 +1,30 @@
+apiVersion: "sparkoperator.k8s.io/v1beta2"
```

Comment:
can we use mock task runner job as default here? nothing prevents us from using Spark Pi but MTR is deterministic with respect to execution time and can be launched with some small sleep time to avoid long waits.

Reply:
good point, but I would suggest to keep using `SparkPi` as a test job because it is faster in comparison with `MockTaskRunner`. In case with the latter, a job needs to fetch an external artifact (> 60 Mb), and that will increase test times, and for the former, the jar is already present inside the image. I've compared two examples locally, `SparkPi` was a way faster then MTR, started with the small arguments' values (1 task 1 second).

Reply:
good point, thanks. we download MTR anyway for testing the killing of running application where we need more control over the duration to avoid flakiness due to timing issues - so the download time is already there.

We also user Spark Pi in DCOS Spark tests to verify Driver output - so leaving it up to you, just wanted to keep the number of simple apps (not targeting any specific corner-cases like shuffle) low.

Reply:
> we download MTR anyway for testing the killing of running application where we need more control over the duration to avoid flakiness due to timing issues - so the download time is already there.

yes, but from my observations during local testing, it won't be reused across different operator instances even in the same namespace from one test run to another.

Reply:
that's right. Spark pulls application jar every time a job is submitted, good point.